### ISSUE TITLE
Data interpretation field for a GenomicAnnotation

### ISSUE THREAD

**Issue by nsheff** [fairtracks/fga-wg#2]
One of the fields for the Annotation entity is the "data interpretation". What do the entities actually represent? For example, a GenomicAnnotation with geometry of "Point" could represent SNPs, or it could represent CpG dinucleotides. A GenomicAnnotation with a geometry of "Region" could represent  "peaks", the results of a peak-calling algorithm, or they could also represent "reads" from a sequencing experiment. (Alternative name possibility: "entity type"?)

There are really probably 2 things here; one is something biological, and the other describes a process.

For example: peak calling: https://bioportal.bioontology.org/ontologies/EDAM?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Foperation_3222

But this is a process; we really needed things called "peaks" which represents the output of a process.

Then for biological things, we need things like "CpGs" or SNPs.

I did find in EDAM the term SNP:
- term is here: https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&lang=en&conceptid=http%3A%2F%2Fedamontology.org%2Fdata_2092&jump_to_nav=true
-  http://edamontology.org/data_2092
- it says obsolete, replace by "Nucleic acid features", which doesn't make sense

So it seems like we would need to make quite a few changes here.

**Comment by nsheff** [comment]
Maybe there are 2 concepts here:
- algorithmic viewpoint: this GenomicAnnotation is the result of a peak-calling algorithm
- biological concept: this GenomicAnnotation (result of a peak-calling algorithm) represents TF binding sites, or chromatin accessibility, or CpG island annotations, or whatever.

So, these are maybe two different types of data interpretation.

**Comment by sveinugu** [comment]
> Maybe there are 2 concepts here:
> 
> * algorithmic viewpoint: this GenomicAnnotation is the result of a peak-calling algorithm
> * biological concept: this GenomicAnnotation (result of a peak-calling algorithm) represents TF binding sites, or chromatin accessibility, or CpG island annotations, or whatever.
> 
> So, these are maybe two different types of data interpretation.

I agree.

In the FAIRtracks schema, the biological concept was mostly captured the `technique` and `target` fields of the [Experiment schema](https://github.com/fairtracks/fairtracks_standard/blob/master/docs/fairtracks_experiment.md), see e.g. this example (from [here](https://github.com/fairtracks/fairtracks_standard/blob/master/json/examples/fairtracks_experiment.example.json)):

```json
{
    "@schema": "https://raw.githubusercontent.com/fairtracks/fairtracks_standard/v1/current/json/schema/fairtracks_experiment.schema.json",
    "global_id": "geo:GSM945229",
    "local_id": "encode:ENCSR000DQP",
    "study_ref": "U54HG004592",
    "sample_ref": "encode:ENCBS192PUU",
    "technique": {
        "term_id": "http://purl.obolibrary.org/obo/OBI_0002017",
        "term_label": "histone modification identification by ChIP-Seq assay"
    },
    "target": {
        "sequence_feature": {
            "term_id": "http://purl.obolibrary.org/obo/SO_0001706",
            "term_label": "H3K4_trimethylation"
        },
        "summary": "H3K4_trimethylation"
    },
    "lab_protocol_description": "https://www.encodeproject.org/documents/8f459e88-6344-434f-8f9f-6375a9ff1880/@@download/attachment/CD20%2B_Stam_protocol.pdf",
    "compute_protocol_description": "https://www.encodeproject.org/documents/6f6351d4-9310-4a3b-a3c2-70ecac47b28b/@@download/attachment/ChIP-seq_Mapping_Pipeline_Overview.pdf"
}
```

Here, the `target` is a specialisation of the concept of `TF binding site`, which is a relation I believe is expressed in the Sequence ontology, in which this term is defined. For other experimental techniques, the target could be e.g. terms like "open chromatin" or "miRNA".

With the biological concept handled by the experiment entitiy, we focused on the algorithmic viewpoint in the [`type_of_condensed_data` field in the Track schema](https://github.com/fairtracks/fairtracks_standard/blob/master/docs/fairtracks_track.md#type_of_condensed_data). The initial suggested values for this field were:

- Sequence-derived regions
- Experimentally-derived regions
- Predicted regions
- Predicted segmentation
- Population-derived variants
- Individual variants
- Peaks
- Broad peaks
- Narrow peaks
- Gapped peaks
- Signal values (fold change)
- Signal values (p-value)
- Signal values (log likelihood)
- Signal values (other)
- Read coverage
- Read counts
- Mapped single-end reads
- Mapped paired-end reads
- Other

**Comment by sunbrn** [comment]
I also agree with the separation made by @nsheff and I can tell that the "GeCo project" aligned with the FAIR track strategy in the sense that the biological perspective was handled in the ExperimentType (we called this target or feature) while the algorithmic one remained expressed in the format of the Item (i.e., Annotation in our new common sense).
However, now I am wondering if... this "algorithmic viewpoint" wouldn't overlap with the "data format" of  #1 (at least, also there we mentioned values such as "Broad peaks"). 
Is there a conceptual difference between the algorithm used to capture an annotation and the format in which its results are encoded? I think so, but then we should be careful when differentiating their value domains.

I really liked Table 2, which is https://doi.org/10.1186/1471-2105-12-494 (@sveinugu paper). That was useful for mapping Data format into Data Geometry. 
Now we need tables that can map:
1. all our allowed values of **Data Geometry** into Data Interpretation (both biological and algorithmic) and
2. 1. all our allowed values of **Data Format** into Data Interpretation (both biological and algorithmic)

Possibly, by doing such an exercise, we will understand if some of these distinctions are superfluous (e.g., because the mapping is always 1:1 for most values)

**Comment by rioualen** [comment]
> There are really probably 2 things here; one is something biological, and the other describes a process.
> 
> For example: peak calling: https://bioportal.bioontology.org/ontologies/EDAM?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Foperation_3222
> 
> But this is a process; we really needed things called "peaks" which represents the output of a process.

Yes, "Peak calling" is an EDAM Operation term (not to be confused with the "Peak detection" Operation term, used in the scope of mass spectrometry).

I think it would be useful to have maybe a "ChIP-seq peak" Data term (maybe as child of "Sequence features" or "Annotation track"). 

However, as I mentioned in yesterday's meeting, we should keep in mind that the concept of ‘peak’, in practice, is often to refer to different things, such as genomic regions (peaks have distinct start and end coordinates), single coordinates (peak summits, peak centers, ...) or bidimentional representations (including signal intensity for example). 


> Then for biological things, we need things like "CpGs" or SNPs.
> 
> I did find in EDAM the term SNP:
> 
> * term is here: https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&lang=en&conceptid=http%3A%2F%2Fedamontology.org%2Fdata_2092&jump_to_nav=true
> * http://edamontology.org/data_2092
> * it says obsolete, replace by "Nucleic acid features", which doesn't make sense
> 
> So it seems like we would need to make quite a few changes here.

The "replaceBy" property doesn't make much sense, indeed. Right now the appropriate term would be [DNA polymorphism](http://edamontology.org/topic_2885), which has "SNP" listed as one of several narrow synonyms.

**Comment by rioualen** [comment]
> Maybe there are 2 concepts here:
> 
> * algorithmic viewpoint: this GenomicAnnotation is the result of a peak-calling algorithm
> * biological concept: this GenomicAnnotation (result of a peak-calling algorithm) represents TF binding sites, or chromatin accessibility, or CpG island annotations, or whatever.
> 
> So, these are may be two different types of data interpretation.

I agree, and that's quite tricky to deal with. This is why many EDAM Format terms have 2 parents: they are organised depending on the "actual" type of format (text, binary, etc), and the "biological" type of format (DNA sequence, etc). However that makes it a bit messy imho, it may not be the best way to approach it. 

Another tricky point is to determine in which main class any concept belongs. For example, the term "[DNA binding sites](http://edamontology.org/topic_3125)" is an EDAM Topic, while the result of a "Peak-calling" Operation is usually some sort of genomic track, set of genomic features, etc (eg Data terms). TF binding sites are not the actual output of the peak-calling process, rather, they can be identified/inferred after more computational and/or human processes (depending on thresholds, biological knowledge, eg many factors that may not be all standardised)

**Comment by sveinugu** [comment]
> Now we need tables that can map:
> 
> 1. all our allowed values of **Data Geometry** into Data Interpretation (both biological and algorithmic) and
> 2. 1. all our allowed values of **Data Format** into Data Interpretation (both biological and algorithmic)
> 
> Possibly, by doing such an exercise, we will understand if some of these distinctions are superfluous (e.g., because the mapping is always 1:1 for most values)

I have now started such a mapping in the spreadsheet: https://docs.google.com/spreadsheets/d/1wqlcnYHwT9yQVCj525n41LQJENbgHeQfn7V572e8ev8/edit?gid=56160221#gid=56160221

I have suggested the other two (or really three) dimensions (geometry, interpretation (algorithmically), interpretation (biological)) starting with file formats as the basis. I have not filled this completely, but enough to draw some conclusions from this exercise:

- There are many-to-many relationships between all four dimensions: File format, Geometry, Interpretation (algo) and Interpretation (bio)
- All of them seems to give valuable metadata
- The split between the two types of interpretation is not always clear-cut, for example relating to variation. While variants relate to several biological mechanisms (mutation, meiosis, ...), the variants themselves are technical/algorithmical concepts where you relate to e.g. a reference sequence. Even worse is this distinction when it comes to concepts like "genes".
- Some of the suggested values for algorithmical interpretations are really hierarchical (different types of peaks, different types of signal values, types of variation, etc)

**Comment by nsheff** [comment]
## Idea: interpretation tags

> The split between the two types of interpretation is not always clear-cut

> There seems to be a need for the schema to allow multiple values for the interpretations,


Given these two things, what do you think of making "interpretations" a bit of a nebulous, tag-style thing?

So, there would be an `interpretations` field, with value as an array of terms. The terms could be, say, from this list of allowable terms, which belong to their own ontology section and could be separated into algorithmic interpretation, biological interpretation, etc.

The advantage of this approach is:

1. nicely handles the possibility to add 0, 1, or more interpretation tags
2. the ontology can handle the ambiguity of the relationships among terms, instead of having that be part of our schema
3. it keeps our schema pretty simple and flexible, so it can handle a lot of these use cases.

**Comment by sveinugu** [comment]
Perhaps our confusion stems from trying to conflate to many different concepts into the term "interpretation"? We have, for instance:

- Types of sequence features contained within this file (GFF type column, Transcription Factor binding site)
- Related biological processes/chemical reactions, e.g. DNA methylation, transcription factor binding
- What values in a file represent (type of measure)
- What is being measured (level of methylation)
- Molecule of interest (DNA, miRNA, ...) 
- Sequence library-related (E.g. ChIP-seq antibody)
- Algorithm-related type of data (Broad peak, Signal)

Can we clear this up somehow? Perhaps other standards might help, for instance [InteroperAble Descriptions of Observable Property Terminology WG (I-ADOPT WG)](https://i-adopt.github.io/)

**Comment by sunbrn** [comment]
> Perhaps our confusion stems from trying to conflate to many different concepts into the term "interpretation"? We have, for instance:
> 
> 1. Types of sequence features contained within this file (GFF type column, Transcription Factor binding site)
> 2. Related biological processes/chemical reactions, e.g. DNA methylation, transcription factor binding
> 3. What values in a file represent (type of measure)
> 4. What is being measured (level of methylation)
> 5. Molecule of interest (DNA, miRNA, ...)
> 6. Sequence library-related (E.g. ChIP-seq antibody)
> 7. Algorithm-related type of data (Broad peak, Signal)
> 
> Can we clear this up somehow? Perhaps other standards might help, for instance [InteroperAble Descriptions of Observable Property Terminology WG (I-ADOPT WG)](https://i-adopt.github.io/)

This is a very good point @sveinugu. Even from points (1) and (2), and their examples "transcription factor binding site" vs. "transcription factor binding", it is clear that we should not confuse "the observed feature" (i.e., signal interpretation) with "the process that led to features to be observed". 

I think the "taxonomy" that you are proposing is very valuable. Possibly, this is where I would place each type of information you mentioned:
(1) as our annotation interpretation, 
(2,3) could be a 2nd-level information in annotation interpretation
(4,5,6) should stay in an Experiment entity
(7) is annotation **format**

If we agree, can we give a precise definition of "annotation interpretation" then? [to be discussed in Feb.4th meeting]

**Comment by sveinugu** [comment]
> I think the "taxonomy" that you are proposing is very valuable. Possibly, this is where I would place each type of information you mentioned:
> (1) as our annotation interpretation,
> (2,3) could be a 2nd-level information in annotation interpretation
> (4,5,6) should stay in an Experiment entity
> (7) is annotation **format**

This is a great start! Some comments:

One could argue that 2 and 4 belongs together, e.g. level of methylation refers to the concept of DNA methylation. This again refers to 5, the molecule of interest, e.g. DNA.

Also 3 (type of measure in file) relates to 4 (what is being measured).

7 is not really a format per se, but perhaps a categorisation of formats, or usage scenario of a format. It also relates to 3 (type of measure).

So these things are interconnected. Still, I think one can delineate as to whether these things relate to data (belongs in `Annotation`), or the experiment/biology (belongs in `Experiment`). Then I think I would slightly change your suggestion and move 2 together with `Experiment`:

Annotation: 1,3,7
Experiment: 2,4,5,6

**Comment by sveinugu** [comment]
> With the biological concept handled by the `Experiment` entity, we focused on the algorithmic viewpoint in the [`type_of_condensed_data` field in the Track schema](https://github.com/fairtracks/fairtracks_standard/blob/master/docs/fairtracks_track.md#type_of_condensed_data). The initial suggested values for this field were:
> 
> * Sequence-derived regions
> * Experimentally-derived regions
> * Predicted regions
> * Predicted segmentation
> * Population-derived variants
> * Individual variants
> * Peaks
> * Broad peaks
> * Narrow peaks
> * Gapped peaks
> * Signal values (fold change)
> * Signal values (p-value)
> * Signal values (log likelihood)
> * Signal values (other)
> * Read coverage
> * Read counts
> * Mapped single-end reads
> * Mapped paired-end reads
> * Other

I just discovered that ENCODE has implemented a similar vocabulary in their file property "Output type" (see under properties in https://www.encodeproject.org/profiles/file). Attached are the currently supported values. I suggest to just adopt this vocabulary instead of spending a large effort incorporating something similar into EDAM. 

[output_type.json](https://github.com/user-attachments/files/21045875/output_type.json)

**Comment by sveinugu** [comment]
Mapping from the 7 aspects of "interpretation" to fields in [v0.9 of the data Model](https://github.com/fairtracks/fga-wg/issues/13):

1. Types of sequence features contained within this file (GFF type column, Transcription Factor binding site)
  - Covered by`GenomicAnnotationFile.sequence_features`
2. Related biological processes/chemical reactions, e.g. DNA methylation, transcription factor binding
  - Covered by`Experiment.biological_processes`
3. What values in a file represent (type of measure)
  - Partly covered by `File.data_content` (ENCODE vocabulary) 
4. What is being measured (level of methylation)
  - Partly covered by `File.data_content` (ENCODE vocabulary) 
5. Molecule of interest (DNA, miRNA, ...)
  - Covered by `Experiment.molecule_type` (from GA4GH Experiment metadata core fields)
6. Sequence library-related (E.g. ChIP-seq antibody)
  - Partly covered by `Experiment.antibody_target` (from GA4GH Experiment metadata extended fields, extensions for other types of assays/sequencing data will be added in the next round)
7. Algorithm-related type of data (Broad peak, Signal)
  - Handled by `File.data_content` (ENCODE vocabulary)

Remaining subissues include:
- [ ] Consider whether values/measures (3 & 4) is well enough covered
- [ ] Follow GA4GH Experiments metadata to add new data-specific fields and contribute if needed
